EIPS/eip-770.md

```diff
@@ -0,0 +1,124 @@
+
+(Step 2) Booker is looking for a resource is querying all resources matching is criteria and is selecting a resource he is willing to reserve
+
+(Step 3.1) In the same operation :
+Booker is approving to RES smart contract the required amount of BTU to make the booking (amount defined in step 1.1 by the provider)
+A booking entry is registered into the smart contract with following information
+
+* Signed hash of resource
+* Signed pointer to all full offer information (datas + metadatas) at the moment of the reservation request. Those information are stored in the offchain booking infrastructure.
+* Public address matching the hash signature
+* desired reservation start date
+* desired reservation end date
+* status : RESERVATION_REQUESTED
+
+(Step 3.2) Booker broadcasts the resource reservation request to the Booking infrastructure adding all end-user profile information off-chain that may be necessary for the reservation approval. This would help comply with privacy protection requirements.
+It is the responsibility of the Booker to notify its end user customer of the full resource information and the provider signature (Availability “snapshot” data). This can be used as proof of reservation details in (the extreme) case resource information at reservation time is deleted from off-chain infrastructure.
+
+(Step 4) Provider is validating the reservation request and broadcasts the new status to “RESERVATION_CONFIRMED”.
+
+(Step 5) RES smart contract is updating the reservation status to “RESERVATION_CONFIRMED”
+
+(Step 6.1 - Optional) Booker cancels the reservation by broadcasting a cancellation request. This triggers a submission to the RES smart contract that empties the registries for this resource.
+Depending on the conditions and cancellation date, the BTU escrowed into the RES smart contract are affected back to the user or to the provider
+
+(Step 6.2 - Optional) Provider cancels the reservation by broadcasting a cancellation request. This triggers a submission to the RES smart contract that empties the registries for this resource. RES smart contract is releasing the escrowed amount to booker.
+
+(Step 7.1 & 7.2) Booker is notifying that the resource has been paid (presumably at resource check-out). RES smart contract is releasing the escrowed BTU back to user in addition to a BTU agreed commision.
+
+## Implementation
+
+The implementation involves following functions :
+
+- publishAvailabilities (likely off-chain) : Publishes one or multiple available resources for reservations and for being “searchable”.
+- listAvailabilities (likely off-chain) : List all available resources matching a search criteria
+- requestReservation : Request for a reservation. The function can be used for initial reservation request OR updating
+- getReservationStatus : Read reservation status. Following values are possible : REQUESTED, REJECTED, CONFIRMED
+
+
+```
+contract ERC770 is ERC20 {
```

Comment:
Why does this need to be an ERC20?

Reply:
Because during the process (step 3.1 or6.1 for instance), the allowance & approve may be called by the RES smart contract

Reply:
The RES contract doesn't need to be an ERC20 contract for calling `allowance` or `approve`.  The callee needs to be ERC20, but the caller doesn't need to.

Reply:
Sorry for my quick reply.
You are right, RES smart contract does not have to be ERC20, however in the sample implementation we are doing it (but not required). Still, the BTU tokens have to ERC20 compliant.
I realize that my description can this code sample is confusion and that it's more clear to separate the token from the RES contract.
This (valuable) feedback will be taken into account in next submission to make things more clear.